hotsos/core/plugins/system/system.py

```diff
@@ -128,6 +129,77 @@ def unattended_upgrades_enabled(self):
 
         return False
 
+    @cached_property
+    def ubuntu_pro_status(self):
+        """Parse and retrieve Ubuntu Pro status
+
+        Returns:
+            Dictionary: Ubuntu pro status as a dictionary, e.g.::
+            {
+                "services": {
+                    "esm-apps": {
+                        "entitled": "yes",
+                        "status": "enabled"
+                    }, /* ... */
+                },
+                "account": "ACME Corporation",
+                "subscription": "Ubuntu Pro (Apps-only) - Virtual",
+                "technical_support_level": "essential",
+                "valid_until": "Sat Jan  1 01:01:01 9999 +03"
+            }
+
+            None: `pro/ua` command not available
+        """
+
+        pro_status = CLIHelper().pro_status()
+        if not pro_status or len(pro_status) < 1:
+            log.info("badness: unable to retrieve `pro status`")
+            return
```

Comment:
Currently this can return a boolean, None or a dictionary which is going to be a little confusing to applications consuming this data. A better approach might be to always return a dict and if upro is not enabled or status not available you could default to:

```
ubuntu-pro: {'status-available': false}
```

And then when status is there you can include ```'status-available': true``` in your dict.